pkg/webhooks/webhooks.go

```diff
@@ -181,3 +183,55 @@ func (wh WebhookHandler) ValidateDelete(hc *v1beta1.HyperConverged) error {
 
 	return nil
 }
+
+func (wh WebhookHandler) HandleMutatingNsDelete(ns *corev1.Namespace, dryRun bool) error {
+	wh.logger.Info("validating namespace deletion", "name", ns.Name)
+
+	operatorNsEnv, err := hcoutil.GetOperatorNamespaceFromEnv()
+	if err != nil {
+		wh.logger.Error(err, "failed to get operator namespace from the environment")
+		return err
+	}
+
+	if ns.Name != operatorNsEnv {
+		wh.logger.Info("ignoring request for a different namespace")
+		return nil
+	}
+
+	ctx := context.TODO()
+	hco := &v1beta1.HyperConverged{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      hcoutil.HyperConvergedName,
+			Namespace: operatorNsEnv,
+		},
+	}
+
+	// TODO: once the deletion of HCO CR is really safe during namespace deletion
+	// (foreground deletion, context timeouts...) try to automatically
+	// delete HCO CR if there.
+	// For now let's simply block the deletion if the namespace with a clear error message
+	// if HCO CR is still there
+
+	key, err := client.ObjectKeyFromObject(hco)
+	if err != nil {
+		wh.logger.Error(err, "failed to get object key for HyperConverged CR")
+	}
+	found := &v1beta1.HyperConverged{}
+	err = wh.cli.Get(ctx, key, found)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			wh.logger.Info("HCO CR doesn't not exist, allow namespace deletion")
+			return nil
+		}
+		wh.logger.Error(err, "failed getting HyperConverged CR")
+		return err
+	}
+
+	return &BlockDeletionWithHyperConvergedError{}
+}
+
+type BlockDeletionWithHyperConvergedError struct{}
+
+func (m *BlockDeletionWithHyperConvergedError) Error() string {
+	return "HyperConverged CR is still present, please remove it before deleting the containing namespace"
+}
```

Comment:
Why do we need this type? if it just returning a string, we can just do:

```golang

	return BlockDeletionWithHyperConvergedError
}

var BlockDeletionWithHyperConvergedError = errors.New("HyperConverged CR is still present, please remove it before deleting the containing namespace")
```

Reply:
done